device/device.yaml

```diff
@@ -17,21 +17,35 @@ components:
       x-include: 
       - '../common/common.yaml#/components/schemas/Named.Object'
       description: >-
-        A container for emulated interfaces and protocol devices.
+        A container for emulated interfaces, loopback interfaces and protocol configurations.
       type: object
-      required: [container_name, ethernet]
       properties:
-        container_name: 
+        ethernets:
           description: >-
-            The unique name of a Port or Lag object that will contain the 
-            emulated interfaces and/or protocol devices.
-          type: string
-          x-constraint:
-          - '/components/schemas/Port/properties/name'
-          - '/components/schemas/Lag/properties/name'
-        ethernet:
+            The emulated Ethernet settings.
+          type: array
+          items:
+            $ref: './ethernet.yaml#/components/schemas/Device.Ethernet'
+        ipv4_loopbacks:
           description: >-
-            The ethernet stack.
-          $ref: './ethernet.yaml#/components/schemas/Device.Ethernet'
-    
-  
+            IPv4 Loopback interface settings attached to an Ethernet interface.
+          type: array
+          items:
+            $ref: './ipv4.yaml#/components/schemas/Device.Ipv4Loopback'
+        ipv6_loopbacks:
+          description: >-
+            IPv6 Loopback interface settings attached to an Ethernet interface.
+          type: array
+          items:
+            $ref: './ipv6.yaml#/components/schemas/Device.Ipv6Loopback'      
+        isis:
+          description: >-
```

Comment:
In top label, we could have array of  instances under device. each instance can have an isis-router. and  each isis-router can have one or more isis-interfaces, route ranges etc. After couple of discussions, we have decided to keep it simple way but can  use more devices to simulate the scenarios.
Benefits of Multi-Instance IS-IS:
You can use multiple IGP instances of IS-IS to redistribute routes among independent IS-IS domains on a single router.
You can construct flexible IS-IS hierarchies across independent IGP domains.
Allows decoupling of multiple IS-IS flooding domains and therefore achieve a more scalable IS-IS deployment.

can the interfaces overlap among multiple instances?   I think the answer is may be no.
https://www.cisco.com/c/en/us/td/docs/routers/xr12000/software/xr12k_r4-0/routing/configuration/guide/rc40xr12k_chapter3.html#con_1202978

Reply:
I think there might be couple of problems with this.
Assume R1 is a test device following this model.
Problem 1 : R1 has two interfaces I1 and I3. Can we have I1 in instance 10 and I3 in instance 20? 
If we allow this, then maybe below problem 2 is applicable -
Problem 2 : R1 has three interfaces I1, I2, I3. 
		I1, I2  belongs to same instance 12. But I1 supports topology (ITID) 1,2. I2 supports topology 2,3. 
		I3 belongs to a different instance 300.
	Now, will this model be able to isolate link-state updates (LSPs) within their interfaces?
-	LSPs of instances 12 should not be forwarded to interface I3. 
-	LSP of instance 12 ITID 1 received on I1 should not be forwarded to I2 or I3.

In general, this model seems different than IxNetwork now. In IxNetwork now we can have only one interface out. There test tool does not make routing decisions. But this model supports multiple interfaces. So it needs to make routing decisions, i.e. which interface to take on way out, and forwarding of LSPs between interfaces. Therefore, I think we should take care of these issues.

N.B. Regarding instance overlapping per interface, you are right it is not supported in Cisco. But RFC 6822 supports it – “ .. When multiple instances share the same circuit, each instance will have a  separate set of adjacencies.” There are other references too.
BTW, I think your model supports it now, since user can configure an array of instances.

Reply:
This has been discussed. If user can create more devices to have more instances. But top level, not keeping ISIS as array to simplify 80% use cases.

Reply:
As discussed, I understand this point was already bounced around, and team thinks supporting one multi instance addresses most usecases. Therefore I am closing this point.